LuaMenu/widgets/gui_battle_room_window.lua

```diff
@@ -144,6 +153,12 @@ local function SetupInfoButtonsPanel(leftInfo, rightInfo, battle, battleID, myUs
 		mapName = battle.mapName:gsub("_", " ")
 		mapName = StringUtilities.GetTruncatedStringWithDotDot(mapName, tbMapName.font, width - 22)
 		tbMapName:SetText(mapName)
+		local mapItem = WG.FeaturedMaps.Get(battle.mapName)
```

Comment:
Do mapItem = WG.FeaturedMaps and WG.FeaturedMaps.Get(battle.mapName) so the code isn't dependent on the API existing. We try to keep the individual widgets somewhat modular.

Reply:
Thanks for the feedback. Can you explain the rationale for this a bit?

What fallback behavior should there be? Display the "<No details>" text? An empty string? Or are you suggesting it revert to the old display entirely?
In the map list, would you expect the fallback to be the code path that I have moved to the new widget, i.e. that it duplicates the logic to call the CommunityWindow API and get the map type string etc (which itself has fallbacks). Or, are you happy for the list to just lose the extra details as if the CommunityWindow API were also absent?

IME having alternate code paths that aren't normally exercised is more likely to cause bugs than lead to graceful degradation. Perhaps there is something I don't understand about what I can expect from widgets though?

Reply:
My rationale is that things should work rather than break in the presence of small bugs. There doesn't need to be any extra handling or alternate code paths since mapItem is already expected to not be a table sometimes. `local mapItem = WG.FeaturedMaps and WG.FeaturedMaps.Get(battle.mapName)` is nil if the map isn't featured or the featured map API doesn't exist.

Reply:
Thanks, makes sense :+1: